safekeeper/src/wal_storage.rs

```diff
@@ -486,11 +489,11 @@ impl Storage for PhysicalStorage {
         // Remove all segments after the given LSN.
         remove_segments_from_disk(&self.timeline_dir, self.wal_seg_size, |x| x > segno).await?;
 
-        let (mut file, is_partial) = self.open_or_create(segno).await?;
+        let (file, is_partial) = self.open_or_create(segno).await?;
 
         // Fill end with zeroes
-        file.seek(SeekFrom::Start(xlogoff as u64)).await?;
-        write_zeroes(&mut file, self.wal_seg_size - xlogoff).await?;
+        file.set_len(xlogoff as u64).await?;
+        file.set_len(self.wal_seg_size as u64).await?;
```

Comment:
We do set_len, so must do sync_all, not fdatasync_file. Otherwise LGTM.

Reply:
Yeah, good call. Thought we might get away with it since the old and new len is the same, but I suppose the kernel might flush metadata in between. Fixed.